src/main/java/ch/cern/eam/wshub/core/services/administration/impl/UserGroupMenuServiceImpl.java

```diff
@@ -0,0 +1,321 @@
+        extMenus.setEXTMENUTYPE(menuType);
+        extMenus.setSEQUENCENUMBER(100); //TODO set to a different number depending on position
+        extMenus.setMOBILE("false");
+
+        // With the request object created, perform the add operation
+       tools.performInforOperation(context, inforws::addExtMenusOp, addExtMenus);
+
+       return "OK";
+    }
+
+    private void validateInputNode(MenuSpecification node) {
+        if (node.path == null || node.menuCode == null || node.userGroup == null) {
+            tools.generateFault("Menu specifications cannot be null");
+        }
+        if (node.path.isEmpty()) {
+            tools.generateFault("Path cannot be empty");
+        }
+        if (node.path.startsWith("/")) {
+            tools.generateFault("Path cannot start with '/'");
+        }
+        if (node.path.endsWith("/")) {
+            tools.generateFault("Path cannot end with '/'");
+        }
+        if (node.path.contains("//")) {
+            tools.generateFault("Path cannot have empty path items");
+        }
+        if (node.path.contains(" ")) {
+            tools.generateFault("Linear Reference ID must be present.");
+        }
+
+        return;
+    }
+
+    private GenericMenuEntry getEntryByPathFromList(List<GenericMenuEntry> menuEntries, String[] words) {
+        GenericMenuEntry current = null;
+
+        Boolean gmeFound = false;
+        for(String next : words) {
+            gmeFound = false;
+            for(GenericMenuEntry gme : menuEntries) {
+                // If the next word in the path is equals to the description of the current item, and the parent item ID
+                // is equals to the current item ID (or is in root hierarchy)
+                if(!gmeFound && gme.description.equals(next) && (current == null || gme.getParent().equals(current.getId()))) {
+                    current = gme;
+                    gmeFound = true;
+                }
+            }
+        }
+
+        return current; // Could be null, if it's root folder (check after method call)
+    }
+
+    private GenericMenuEntry findGMEFunctionId(GenericMenuEntry entryToDelete, List<GenericMenuEntry> menuEntries, String functionId) {
+        for (GenericMenuEntry gme : menuEntries) {
+            if (entryToDelete.getId().equals(gme.getParent()) && // If the parent of the current gme is entry to delete
+                gme.getFunctionId().equals(functionId)) {  // If the function is the one we want to delete
+                return gme;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Deletes a function item, or a menu item with all its children.
+     * @param context the user credentials
+     * @param node the specified path and function to delete, for a specific user group
+     * @return
+     */
+    @Override
+    public String deleteFromMenuHierarchy(InforContext context, MenuSpecification node) throws InforException {
+        this.validateInputNode(node);
```

Comment:
This validation method should be functional and, as such, be a static method

Reply:
Solved.